http/src/v0/pin.rs

```diff
@@ -0,0 +1,42 @@
+use futures::future::join_all;
+use crate::v0::support::{with_ipfs};
+use ipfs::{Cid, Ipfs, IpfsTypes, Multiaddr};
+use serde::{Deserialize, Serialize};
+use warp::{reply, Rejection, Reply, Filter, query, path};
+
+#[derive(Debug, Deserialize)]
+struct AddRequest {
+    arg: Vec<Cid>,
```

Comment:
Just as a small tactical heads up, the `serde_urlencode` nor `serde_qs` support these kind of arguments (multiple same names aggregated under a single type); the parsing has to be implemented manually at the moment as was for the https://github.com/ipfs-rust/rust-ipfs/blob/master/http/src/v0/refs/options.rs.

For `serde_urlencode` I found some closed issue that this is out of scope. Having looked at it for a while, I don't think the serde implementation for this would be too easy or then I am just not understanding something of serde.

Doing this manually though, is quite simple. It also removes the need create remote wrappers for Cid, and so on, while being a bit non-idiomatic.